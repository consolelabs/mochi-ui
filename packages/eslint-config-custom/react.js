const { resolve } = require('node:path')

const project = resolve(process.cwd(), 'tsconfig.json')

const dfLintConfig = require.resolve('@dwarvesf/react-eslint-config')

module.exports = {
  extends: [dfLintConfig, 'eslint-config-turbo'],
  plugins: ['prettier'],
  parserOptions: {
    project,
  },
  globals: {
    JSX: true,
  },
  ignorePatterns: [
    'node_modules/',
    'dist/',
    '.eslintrc.js',
    '**/*.css',
    '*.snap',
  ],
  settings: {
    'import/resolver': {
      typescript: {
        project,
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
  rules: {
    'react/function-component-definition': 'off',
    'turbo/no-undeclared-env-vars': 'off',
    'react/jsx-key': 'off',
    'no-dupe-class-members': 'off',
    'prefer-arrow-callback': 'off',
    'import/export': 'off',
    'prettier/prettier': [
      'error',
      {
        bracketSpacing: true,
        jsxBracketSameLine: false,
        jsxSingleQuote: false,
        printWidth: 80,
        proseWrap: 'always',
        semi: false,
        singleQuote: true,
        tabWidth: 2,
        trailingComma: 'all',
        xmlWhitespaceSensitivity: 'ignore',
      },
    ],
  },
  overrides: [
    {
      parser: '@typescript-eslint/parser',
      files: ['**/*.ts', '**/*.tsx'],
      plugins: ['@typescript-eslint'],
      rules: {
        'no-use-before-define': 'off',
        'react/react-in-jsx-scope': 'off',
        'react/jsx-sort-props': 'off',
        'no-void': 'off',
        'no-unused-vars': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_',
            caughtErrorsIgnorePattern: '^_',
          },
        ],
        '@typescript-eslint/no-use-before-define': [
          'error',
          { functions: false, classes: false },
        ],
      },
    },
  ],
}
