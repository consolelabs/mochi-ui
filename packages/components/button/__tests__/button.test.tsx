import { fireEvent, render } from '@testing-library/react'
import Button from '../src/button'

describe('Button', () => {
  it('renders the button with the correct text', () => {
    const buttonText = 'Click me'
    const { getByText } = render(<Button>{buttonText}</Button>)
    const button = getByText(buttonText)
    expect(button).toBeInTheDocument()
  })

  it('calls the onClick function when clicked', () => {
    const onClick = jest.fn()
    const { getByText } = render(<Button onClick={onClick}>Click me</Button>)
    const button = getByText('Click me')
    fireEvent.click(button)
    expect(onClick).toHaveBeenCalled()
  })

  it('disables the button when loading', () => {
    const { container } = render(<Button loading>Click me</Button>)
    const button = container.querySelector('button')
    expect(button).toHaveClass('pointer-events-none')
  })

  it('matches snapshot', () => {
    const { container } = render(<Button>Click me</Button>)
    expect(container).toMatchSnapshot()
  })

  it('matches snapshot when loading', () => {
    const { container } = render(<Button loading>Click me</Button>)
    expect(container).toMatchSnapshot()
  })
})
