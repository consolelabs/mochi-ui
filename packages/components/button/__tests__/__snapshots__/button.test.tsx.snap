// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    class="flex items-center gap-x-2 font-semibold transition shadow-button border text-sm h-10 leading-4 px-4 py-[9px] rounded-lg bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Button matches snapshot when loading 1`] = `
<div>
  <button
    class="flex items-center gap-x-2 font-semibold transition shadow-button border text-sm h-10 leading-4 pointer-events-none px-4 py-[9px] rounded-lg bg-primary-700 text-white border-transparent hover:bg-primary-800 focus:shadow-button-focused-primary disabled:bg-primary-400"
    type="button"
  >
    <div
      class="flex items-center h-5 text-base"
    >
      <svg
        class="text-[40px]"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="18"
          cy="12"
          fill="currentColor"
          r="0"
        >
          <animate
            attributeName="r"
            begin="0.67"
            calcMode="spline"
            dur="1.5s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;2;0;0"
          />
        </circle>
        <circle
          cx="12"
          cy="12"
          fill="currentColor"
          r="0"
        >
          <animate
            attributeName="r"
            begin="0.33"
            calcMode="spline"
            dur="1.5s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;2;0;0"
          />
        </circle>
        <circle
          cx="6"
          cy="12"
          fill="currentColor"
          r="0"
        >
          <animate
            attributeName="r"
            begin="0"
            calcMode="spline"
            dur="1.5s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;2;0;0"
          />
        </circle>
      </svg>
    </div>
  </button>
</div>
`;
